registry/strings/functions.go

```diff
@@ -682,6 +682,52 @@ func (sr *StringsRegistry) SwapCase(str string) string {
 	}, str)
 }
 
+// Capitalize capitalizes the first letter of 'str'.
+//
+// Parameters:
+//
+//	str string - the string to capitalize.
+//
+// Returns:
+//
+//	string - the string with the first letter capitalized.
+//
+// Example:
+//
+//	{{ "hello world" | capitalize }} // Output: "Hello world"
+func (sr *StringsRegistry) Capitalize(str string) string {
+	for i, r := range str {
+		if unicode.IsLetter(r) {
+			return str[:i] + string(unicode.ToUpper(r)) + str[i+1:]
+		}
```

Comment:
Are you reinventing strings.Title?

https://pkg.go.dev/strings#Title

Now replaced by golang.org/x/text/cases

https://pkg.go.dev/golang.org/x/text/cases#Title

Reply:
Got it

The test helped me to understand: it might be interesting when the leading characters are not letter.

I would recommend you to test Title in x/text because it may have interesting option to do it, or it simply work out of the box

Reply:
`strings.Title` are used in `toTitleCase` and don't give us the same behavior

Reply:
Oh indeed, it changes the first letter of all words.

Reply:
I edited my previous message 👆, and I see you reacted, please read the last version if you didn't

Reply:
This is my first choice, and I did some research on other languages to find the correct naming. Capitalize is commonly used in several languages for this purpose:

Sources :
- Python: https://www.w3schools.com/python/ref_string_capitalize.asp
- Ruby: https://apidock.com/ruby/String/capitalize
- java (trough package) : https://commons.apache.org/proper/commons-lang/javadocs/api-2.5/org/apache/commons/lang/StringUtils.html#capitalize%28java.lang.String%29

Thanks for the second ping

Reply:
OK then, why not

Wording is a pain 😅

Reply:
True 😅